src/components/History/History.vue

```diff
@@ -270,13 +268,14 @@ export default {
       }
     },
     comeBackTo(historyElement) {
-      if (this.couldLoad) {
-        this.$router.push({
-          path: `/selector/${historyElement.decade}`,
-          query: { history: true, imageId: historyElement.imageId },
-        });
-        this.$emit("closeFullHistory");
-      }
+      this.$router.push({
+        path: `/canvas`,
+        query: {
+          imageId: JSON.stringify(historyElement.imageId),
+          tag: JSON.stringify(historyElement.tag),
```

Comment:
Why do you need to use JSON.stringify?

Reply:
For tag, it's mandatory because some tag has '/' char and without the stringify, the url is wrong. With the current data, I think it's not necessary for imageId because it's an integer but I don't know for next data. To be more secure, I add it also for imageId but I can remove it.

Reply:
`encodeURIComponent` would make more sense in that case for both, not sure JSON.stringify would always convert '/' actually, not supposed to

Reply:
Ok, in my side, during a test, that works but yeah encode and decode are better so I change the code to implement this way.